packages/itwinui-react/src/core/Tabs/Tabs.tsx

```diff
@@ -64,547 +43,759 @@ type TabsOrientationProps =
-  onTabSelected?: (index: number) => void;
-  /**
-   * Index of the active tab.
+   * Color of the bar on the active tab.
+   * @default 'blue'
    */
-  activeIndex?: number;
+  color?: 'blue' | 'green';
   /**
    * Control whether focusing tabs (using arrow keys) should automatically select them.
    * Use 'manual' if tab panel content is not preloaded.
    * @default 'auto'
    */
   focusActivationMode?: 'auto' | 'manual';
   /**
-   * Color of the bar on the active tab.
-   * @default 'blue'
-   */
-  color?: 'blue' | 'green';
-  /**
-   * Custom CSS class name for tabs.
+   * Value of the tab that should be active on initial render.
+   *
+   * Should be used for uncontrolled state (when no `value` passed).
+   *
+   * If not specified, then first tab will be active by default.
    */
-  tabsClassName?: string;
+  defaultValue?: string;
   /**
-   * Custom CSS class name for tab panel.
+   * Value of the active tab for controlled state.
    */
-  contentClassName?: string;
+  value?: string;
   /**
-   * Custom CSS class name for the tabs wrapper.
+   * Function that gets called when active tab is changed.
+   *
+   * Should be used alongside `activeValue` prop.
    */
-  wrapperClassName?: string;
+  onValueChange?: (value: string) => void;
   /**
-   * Content inside the tab panel.
+   * @deprecated Do not use.
    */
-  children?: React.ReactNode;
-} & TabsOrientationProps &
-  TabsTypeProps &
-  TabsOverflowProps;
-
-/**
- * Tabs organize and allow navigation between groups of content that are related and at the same level of hierarchy.
- * @example
- * const tabs = [
- *   <Tab label='Label 1' />,
- *   <Tab label='Label 2' />,
- *   <Tab label='Label 3' />,
- * ];
- * <Tabs labels={tabs} />
- *
- * @example
- * <Tabs orientation='vertical' labels={tabs} />
- *
- * @example
- * const tabsWithSublabels = [
- *   <Tab label='Label 1' sublabel='First tab' />,
- *   <Tab label='Label 2' sublabel='Active tab' />,
- * ];
- * <Tabs labels={tabsWithSublabels} activeIndex={1} />
- *
- * @example
- * const tabsWithIcons = [
- *   <Tab label='Label 1' icon={<SvgPlaceholder />} />,
- *   <Tab label='Label 2' icon={<SvgPlaceholder />} />,
- * ];
- * <Tabs labels={tabsWithIcons} type='pill' />
- */
-export const Tabs = (props: TabsProps) => {
-  // Separate actions from props to avoid adding it to the DOM (using {...rest})
-  let actions: Array<React.ReactNode> | undefined;
-  if (props.type !== 'pill' && props.actions) {
-    actions = props.actions;
-    props = { ...props };
-    delete props.actions;
-  }
-  // Separate overflowOptions from props to avoid adding it to the DOM (using {...rest})
-  let overflowOptions: OverflowOptions | undefined;
-  if (
-    props.type !== 'borderless' &&
-    props.type !== 'pill' &&
-    props.overflowOptions
-  ) {
-    overflowOptions = props.overflowOptions;
-    props = { ...props };
-    delete props.overflowOptions;
-  }
+  defaultChecked?: never; // To remove `defaultChecked` from `<div>` props.
```

Comment:
Can make this a TODO comment for visibility

Reply:
It's not a todo item. This prop IS removing defaultChecked from div props. Maybe comment needs to be updated to full sentence.

Reply:
Is there a reason to not [`Omit<>`](https://www.typescriptlang.org/docs/handbook/utility-types.html#omittype-keys) it instead of deprecating it but allowing to pass it?

Reply:
I don't remember. @mayank99 any insights?

Reply:
you can try omitting it, i think it will be added back by `<div>` props (via `PolymorphicForwardRefComponent`

Reply:
the comments could be reworded as "Removing…" instead of "To remove…"

Reply:
Updated comments